### DISCUSSION TITLE
Building Coqui with TFLite delegate support

### DISCUSSION THREAD

**Discussion by jeremiahrose** [coqui-ai/STT#1925] [Q&A]
Hi there!

I gather from the [docs](https://stt.readthedocs.io/en/latest/BUILDING.html) and some old [Deepspeech discussions](https://github.com/mozilla/DeepSpeech/issues/2270) that building and using Coqui STT with the GPU, NNAPI and Hexagon delegates was still in the experimental phase around 1 year ago (early 2020).

I'm wondering where that is at now, and what I would need to do to get it working?

Specifically I want to know the proper way to:

- Build `libstt.so`, `stt` native client, with delegates enabled. What Bazel flags are required? (I'm cross-compiling to ARMv7 Linux)
- Build the delegates themselves. What Bazel targets are required, and how do I make sure they're statically linked into `libstt.so`?
- Run `stt` native client with a delegate (is e.g `STT_TFLITE_DELEGATE=gpu` all that's required?)
- Make the Coqui 0.9.3 .tflite model work with a delegate (GPU or Hexagon, probably)

Happy to do whatever debugging is required to get this to work. Also hoping that this thread might become a starting point for other people who are trying to get this working.

**Comment by reuben** [comment]
All you need to do is set the environment variable and verify the model is behaving as intended. No need to change the build (or even build yourself).

> Run stt native client with a delegate (is e.g STT_TFLITE_DELEGATE=gpu all that's required?)

Yep. Code here: https://github.com/coqui-ai/STT/blob/df26eca4d20fe7ac9eda5e55e86a2e1ef66bb745/native_client/tflitemodelstate.cc#L102-L156

**Reply by jeremiahrose** [reply]
Thanks for the reply Reuben.
We are cross-compiling Coqui within a Yocto build system so fetching pre-built binaries is not currently an option.
